### PR TITLE
Normalize contractions

### PR DESCRIPTION
**Pull request by kimt33**
Working off of PR #42 

What:
1. Add `assign_norm_cont` to `ContractedCartesianGaussians`
2. Add attribute `norm_cont` for storing normalization constants for the contractions
3. Change name of property from `norm` to `norm_prim`
4. Make appropriate changes to tests and API
5. Fix normalization of spherical contractions

Since contraction coefficients are not always such that the contractions are normalized after normalizing the primitives, they need to be normalized. This means that they must access the `overlap` module, which introduces a bit of circular import. We can avoid circular import error by placing the import statement within a function, so the method `assign_norm_cont` to assign the attribute `norm_cont`. Normalization constants are stored to avoid recomputing it. 

Also, the transformation from cartesian contractions to spherical contractions was not converting the normalization constant. This was fixed.

### CODE REVIEW

**Comment by kimt33** [comment]
[Add function for evaluating normalization constant for spherical Gaussian](https://trello.com/c/aq9rp2Ma/9-add-function-for-evaluating-normalization-constant-for-spherical-gaussian)

**Comment by kimt33** [comment]
[Add function for normalizing `ContractedCartesianGaussians`](https://trello.com/c/aIR0krb2/47-add-function-for-normalizing-contractedcartesiangaussians)

File: tests/test_overlap.py [hunk 1]
```diff
@@ -109,3 +109,69 @@ def test_overlap_basis_spherical_lincomb():
...
+        test_basis = f.read()
+    basis_dict = parse_nwchem(test_basis)
+
+    basis = make_contractions(basis_dict, ["Kr"], np.array([[0, 0, 0]]))
+    overlap_obj = Overlap(basis)
+    assert np.allclose(np.diag(overlap_obj.construct_array_cartesian()), 1)
+
+
+def test_overlap_basis_spherical_norm_sto6g():
+    """Test the norm of gbasis.eval.overlap_basis_spherical on the ANO-RCC basis set.
```
Comment (wilhadams) [inline tests/test_overlap.py]: This one is STO-6G not ANO-RCC

File: tests/test_overlap.py [hunk 2]
```diff
@@ -109,3 +109,69 @@ def test_overlap_basis_spherical_lincomb():
...
+    overlap_obj = Overlap(basis)
+    assert np.allclose(np.diag(overlap_obj.construct_array_cartesian()), 1)
+
+    basis = make_contractions(basis_dict, ["Xe"], np.array([[0, 0, 0]]))
+    overlap_obj = Overlap(basis)
+    assert np.allclose(np.diag(overlap_obj.construct_array_cartesian()), 1)
+
+
+def test_overlap_basis_cartesian_norm_sto6g():
+    """Test the norm of gbasis.eval.overlap_basis_cartesian on the ANO-RCC basis set.
```
Comment (wilhadams) [inline tests/test_overlap.py]: STO-6G not ANO-RCC

File: tests/test_overlap.py (hunk 1 above)
Comment (kimt33) [reply tests/test_overlap.py]: Thanks!

File: tests/test_overlap.py (hunk 2 above)
Comment (wilhadams) [reply tests/test_overlap.py]: It still says ANO-RCC not STO-6G in the first line

**Review by wilhadams — CHANGES_REQUESTED** [review]

**Pull request merged**